examples/sp/sp_tutorial.py

```diff
@@ -0,0 +1,312 @@
+# add noise to one of them.
+	
+inputX1 = np.zeros(inputSize, dtype=uintType)
+inputX2 = np.zeros(inputSize, dtype=uintType)
+outputX1 = np.zeros(columnNumber, dtype=uintType)
+outputX2 = np.zeros(columnNumber, dtype=uintType)
+
+for i in range(inputSize):
+  inputX1[i] = random.randrange(2)
+
+x = []
+y = []   
+for noiseLevel in np.arange(0, 1.1, 0.1):
+  resetVector(inputX1, inputX2)
+  corruptVector(inputX2, noiseLevel)
+  x.append(noiseLevel)
+  y.append(pctOverlap(inputX1, inputX2, inputSize))
+
+print ""
+print "---------------------------------"
+print "Figure 2 shows the input overlap between 2 identical binary"
+print "vectors in function of the noise applied to one of them."
+print "0 noise level means that the vector remains the same, whereas"
+print "1 means that the vector is the logical negation of the original"
+print "vector."
+print "The relationship between overlap and noise level is practically"
+print "linear and monotonically decreasing."
+print "---------------------------------"
+print ""
+
+plt.plot(x, y)
+plt.xlabel("Noise level")
+plt.ylabel("Input overlap")
+plt.title("Figure 2")
+plt.show()
+
+# Part 2b: 
+# -------
+# The output overlap between two binary input vectors is the overlap of the
+# columns that become active once they are fed to the SP. In this part we
+# turn learning off, and observe the output of the SP as we input two binary 
+# input vectors with varying level of noise.
+# Starting from two identical vectors (that yield the same active columns)
+# we would expect that as we add noise to one of them their output overlap
+# decreases.
+# In this part we will show how the output overlap behaves in function of the
+# input overlap between two vectors.
+
+x = []
+y = []   
+for noiseLevel in np.arange(0, 1.1, 0.1):
+  resetVector(inputX1, inputX2)
+  corruptVector(inputX2, noiseLevel)
+  
+  sp.compute(inputX1, False, outputX1)
+  sp.compute(inputX2, False, outputX2)  
+  
+  x.append(pctOverlap(inputX1, inputX2, inputSize))
+  y.append(pctOverlap(outputX1, outputX2, columnNumber))
+
+print ""
+print "---------------------------------"
+print "Figure 3 shows the output overlap between two sparse representations" 
+print "in function of their input overlap. Starting from two identical binary vectors"
+print "(which yield the same active columns) we add noise two one of them"
+print "feed it to the SP, and estimate the output overlap between the two"
+print "representations in terms of the common active columns between them."
+print "As expected, as the input overlap decrease, so does the output overlap."
+print "However, this relationship is sublinear, implying that the SP is very"
+print "sensitive to noise."
+print "---------------------------------"
+print ""
+
+plt.plot(x, y)
+plt.xlabel("Input overlap")
+plt.ylabel("Output overlap")
+plt.title("Figure 3")
+plt.show()
+
+# Part 3: 
+# -------
+# As we saw in the last section, an untrained SP is very sensitive to noise in a binary vector
```

Comment:
Similar to my earlier comment, I would take out the first sentence. Focus on the fact that with training, the SP can become less sensitive to noise.